sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.service.ts

```diff
@@ -108,4 +134,152 @@ export class EducationProgramOfferingControllerService {
       description: getOfferingNameAndPeriod(offering),
     }));
   }
+
+  /**
+   * Creates the offering model to be validated and saved using the DTO received.
+   * @param institutionId institution id.
+   * @param locationId location id.
+   **The authorization to validated if the location belongs to the institution
+   **is not expected to be performed by this method.
+   * @param programId program id.
+   * @param payload information to generate the model to perform the offering
+   * validation and persistence.
+   * @returns offering model to be validated and saved.
+   */
+  async mapOfferingAPIInDTOTOSaveOfferingModel(
+    institutionId: number,
+    locationId: number,
+    programId: number,
+    payload: EducationProgramOfferingAPIInDTO,
+  ): Promise<SaveOfferingModel> {
+    // Program information required to perform the offering validations.
+    const program = await this.programService.getEducationProgramDetails(
+      programId,
+      institutionId,
+    );
+    if (!program) {
+      throw new NotFoundException(
+        "Program to create the offering not found for the institution.",
+      );
+    }
+    return {
+      ...payload,
+      locationId,
+      studyBreaks: payload.breaksAndWeeks?.studyBreaks,
+      programContext: program,
+    };
+  }
+
+  /**
+   * Verify if all CSV validations were performed with success and throw
+   * a BadRequestException in case of some failure.
+   * @param csvValidations validations to be verified.
+   */
+  assertCSVValidationsAreValid(csvValidations: OfferingCSVValidationResult[]) {
```

Comment:
IMO assertCSVValidations itself explains the context of the method.

Reply:
Yes, I can see it as a valid point.
The idea here, following the same from test frameworks, when we execute an "assert" we "assert" something, that is why I added the "AreValid" in the end.
If it is a recommendation only I will keep as it is 😉